src/Microsoft.ML.Data/MLContext.cs

```diff
@@ -61,7 +61,7 @@ public sealed class MLContext : IHostEnvironment
         /// <summary>
         /// The handler for the log messages.
         /// </summary>
-        public Action<string> Log { get; set; }
+        public event EventHandler<string> Log;
```

Comment:
I think we should be using an `EventArgs` type instead of `string`.

`public event EventHandler<LoggingEventArgs> Log;`

Reply:
See https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/event

Reply:
That's a good point. It's been quite a while since I had to create a custom event, it slipped my mind. I'll update the PR.